include/highfive/H5DataType.hpp

```diff
@@ -158,6 +158,67 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+///
+/// \brief Create a enum HDF5 datatype
+///
+/// \code{.cpp}
+/// enum class Position {
+///     FIRST = 1,
+///     SECOND = 2,
+/// };
+///
+/// EnumType<Position> create_enum_position() {
```

Comment:
We have to explain better the code, unfortunately it's not so self-descriptive, specially the specialization.
And btw, I assume this specialization must be done within the HighFive namespace. I would stress that in the comments.
I would really try to reduce the amount of code, maybe drop the function, chain calls... e.g.:
`file.createDataset("/foo", DataSpace(1), en).write(Position::FIRST);`
Can we use the shortcut syntax?
`file.createDataset("/foo", en, Position::FIRST);`